### ISSUE TITLE
Meilisearch Plugin: Updated products omits categories in transformer function

### ISSUE THREAD

**Issue by hyenabyte** [medusajs/medusa#5243]
### Describe the bug

I have a transformer function set up to do some basic mapping of data, looks like this:
```js
{
  transformer: (product) => {
    return {
      id: product.id,
      title: product.title,
      handle: product.handle,
      thumbnail: product.thumbnail,
      created_at: product.created_at,
      collection: product.collection.handle,
      price: product.variants[0]?.prices[0]?.amount ?? 0,

      categories: product.categories,
      variants: product.variants,
    };
  },
}
```
I noticed that categories would go missing from the indexed data once I updated the product, and I figured out if I logged out `product` categories would be included at the first indexing on server start, but when the transform was run again on product update categories are no longer included.

### System information

- Medusa version (including plugins):
  - Medusa version: 1.3.7
  - medusa-plugin-meilisearch version: 2.0.10
  - Meili version: 1.3.1
- Node.js version: v18.17.1
- Database: postgres
- Operating system: MacOS 13.5.2
- Browser (if relevant):

### Steps to reproduce the behavior

1. Create a product and assign it one or more categories
2. Restart the server and have the product indexed
3. Edit the product, i.e. change its name
4. Observe the missing categories in meilisearch for the updated product

### Expected behavior

The data should not differ between create and update in the transform function, or alternatively there should be a way to easily discern between create, update, delete and so on within the function.

### Additional context

The entire config for the plugin:
```js
  {
    resolve: `medusa-plugin-meilisearch`,
    options: {
      config: {
        host: process.env.MEILISEARCH_HOST ?? 'localhost:7700',
        apiKey: process.env.MEILISEARCH_API_KEY ?? '',
      },
      settings: {
        products: {
          indexSettings: {
            searchableAttributes: ['title', 'description', 'variant_sku', 'categories'],
            displayedAttributes: [
              'id',
              'title',
              'description',
              'variant_sku',
              'thumbnail',
              'handle',
              'categories',
              'variants',
              'variantsActual',
              'collection',
              'price',
            ],
            sortableAttributes: ['title', 'created_at', 'price'],
            filterableAttributes: ['categories', 'collection', 'variants'],
          },
          primaryKey: 'id',
          transformer: (product) => {
            // console.log('Transformer run for product: ', product.id);
            // console.log(product);
            return {
              id: product.id,
              title: product.title,
              handle: product.handle,
              thumbnail: product.thumbnail,
              created_at: product.created_at,
              collection: product.collection.handle,

              price: product.variants[0]?.prices[0]?.amount ?? 0,

              categories: product.categories,

              variants: product.variants,
            };
          },
        },
      },
    },
  },
```

**Comment by nfzv** [comment]
Hey, having the same issue, but it took me a month to realize/debug that categories went indeed missing on updates. Any progress on that?

**Comment by nfzv** [comment]
This PR probably resolves the issue: #6555

**Comment by nfzv** [comment]
Patched the PR changes #6555  locally, can confirm it fixes this issue.

**Comment by sradevski** [comment]
Hey, thanks for the report! Since v2 brought a lot of architectural and API changes on the backend, we will be closing this ticket since it no longer applies to our new setup, or the issue has already been fixed. If you are still facing issues with v1,
please open a new ticket and we will address it as soon as possible. Thanks! 🙏
